components/script/dom/bluetoothadvertisingevent.rs

```diff
@@ -0,0 +1,127 @@
+use dom::event::{Event, EventBubbles, EventCancelable};
+use dom::globalscope::GlobalScope;
+use dom::window::Window;
+use servo_atoms::Atom;
+
+// https://webbluetoothcg.github.io/web-bluetooth/#bluetoothadvertisingevent
+#[dom_struct]
+pub struct BluetoothAdvertisingEvent {
+    event: Event,
+    device: MutHeap<JS<BluetoothDevice>>,
+    name: Option<DOMString>,
+    appearance: Option<u16>,
+    tx_power: Option<i8>,
+    rssi: Option<i8>,
+}
+
+impl BluetoothAdvertisingEvent {
+    pub fn new_inherited(device: &BluetoothDevice,
+                         name: Option<DOMString>,
+                         appearance: Option<u16>,
+                         tx_power: Option<i8>,
+                         rssi: Option<i8>)
+                         -> BluetoothAdvertisingEvent {
+        BluetoothAdvertisingEvent {
+            event: Event::new_inherited(),
+            device: MutHeap::new(device),
+            name: name,
+            appearance: appearance,
+            tx_power: tx_power,
+            rssi: rssi,
+        }
+    }
+
+    pub fn new(global: &GlobalScope,
+               type_: Atom,
+               bubbles: EventBubbles,
+               cancelable: EventCancelable,
+               device: &BluetoothDevice,
+               name: Option<DOMString>,
+               appearance: Option<u16>,
+               txPower: Option<i8>,
+               rssi: Option<i8>)
+               -> Root<BluetoothAdvertisingEvent> {
+        let ev = reflect_dom_object(box BluetoothAdvertisingEvent::new_inherited(device,
+                                                                                 name,
+                                                                                 appearance,
+                                                                                 txPower,
+                                                                                 rssi),
+                                    global,
+                                    BluetoothAdvertisingEventBinding::Wrap);
+        {
+            let event = ev.upcast::<Event>();
+            event.init_event(type_, bool::from(bubbles), bool::from(cancelable));
+        }
+        ev
+    }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothadvertisingevent-bluetoothadvertisingevent
+    pub fn Constructor(window: &Window,
+                       type_: DOMString,
+                       init: &BluetoothAdvertisingEventInit)
+                       -> Fallible<Root<BluetoothAdvertisingEvent>> {
+        let global = window.upcast::<GlobalScope>();
+        let device = init.device.r();
+        let name = init.name.clone();
+        let appearance = init.appearance.clone();
+        let txPower = init.txPower.clone();
+        let rssi = init.rssi.clone();
+        let bubbles = EventBubbles::from(init.parent.bubbles);
+        let cancelable = EventCancelable::from(init.parent.cancelable);
+        let event = BluetoothAdvertisingEvent::new(global,
```

Comment:
Do we need the variable binding here, can't we just return `Ok(BluetoothAdvertisingEvent::new(...` ?